cirq/sim/density_matrix_utils.py

```diff
@@ -257,3 +257,12 @@ def _validate_indices(num_qubits: int, indices: List[int]) -> None:
     if any(index >= num_qubits for index in indices):
         raise IndexError('Out of range indices, must be less than number of '
                          'qubits but was {}'.format(indices))
+
+
+def von_neumann_entropy(density_matrix: np.ndarray) -> np.ndarray:
+    """Calculates von Neumann entropy of density matrix"""
+    if 0 in density_matrix:
+        raise ValueError('Cannot calculate von Neumann entropy of a density '
```

Comment:
Errr... I think there's been a misunderstanding. You are using per-entry arithmetic instead of whole-matrix arithmetic.

Let V be the list of eigenvalues of D. Then then von neumann entropy of D is -\sum_k V_k \log_2 V_k.

There is no constraint related to entries in the matrix having to be non-zero.

Reply:
Ah! Yes.

Reply:
Today i learned: https://en.wikipedia.org/wiki/Logarithm_of_a_matrix